### PR TITLE
[ALE-07] Refactor  API logic to use entities

### PR DESCRIPTION
**Pull request by alejandrobenjumeaGL**

### CODE REVIEW

**Review by julianfigueroaGL — APPROVED** [review]
Looks good to me

File: src/api/index.js [hunk 1]
```diff
@@ -59,25 +81,27 @@ class Api {
...
-    const response = await fetch(
-      `${API_ENDPOINT}/search?${parsedQueryString}`,
-    );
-    const { results } = await response.json();
-    return filter(propEq('isStreamable', true), results);
+    return compose(
+      this.mapFavoritedProperty,
+      isStreamable,
+      take(limit), // Helpfull when data is mocked
+    )(results);
```
Comment (josepenagosGL) [inline src/api/index.js:99]: Question just to clarify. When data is mocked, this is going to grab the number of records based on the limit. When not mocked, the amount of records within results will be the same as limit. Am I right?

File: src/api/index.js [hunk 2]
```diff
@@ -34,9 +49,16 @@ import {
  */
 
 class Api {
+  constructor() {
+    this._state = readLocalStorage(KeyNames.state, DEFAULT_STATE);
+
+    this.mapFavoritedProperty = map((track) => assoc('isFavorited', this.isFavorited(track), track));
```
Comment (josepenagosGL) [inline src/api/index.js:55]: Why is this function defined as a property of the class rather than a regular function?

Comment (alejandrobenjumeaGL) [reply src/api/index.js:55]: **isFavorited** function needs to know about the favorite list, so in line 295 you can see that this info is in the instance, that's why we need it in that way

File: src/api/index.js (hunk 1 above)
Comment (alejandrobenjumeaGL) [reply src/api/index.js:99]: Yes your right, this was to improve readability. Other approach could be something like 
`ifElse(
        isMocked,
        take(limit),
        identity,
      )`,
but I think, for the purpose of the amount of data, is enough having it simple

**Review by opolancoh-gl — APPROVED** [review]
Changes approved

**Pull request merged**